### DISCUSSION TITLE
How to globally configure `KubernetesPodOperator` settings?

### DISCUSSION THREAD

**Discussion by ketozhang** [apache/airflow#36821] [Q&A] [answered]
Although most settings can be configured with an equivalent pod template, there are others that can't (e.g.,`startup_timeout_seconds`, `log_events_on_failure`) at least without a lot more work (e.g., adding a startup probe).

Can this be configured globally? 

Only idea I have is to subclass `KubernetesPodOperator`

**Comment by potiuk** [comment, marked as answer]
You can use `default_args` in your DAGs and set value in those from a common imported constant. That's as close as it can get to setting "global" value.

**Reply by ketozhang** [reply]
Darn, I hope that wasn't the case. From an admin's point-of-view, we'd want all KPOs to turn on `log_events_on_failure` without having all DAG devs edit their `default_args`. So the workaround would be custom operators or define the `default_args` constant somewhere for devs to inject.

**Reply by Taragolis** [reply]
In addition you could use [Cluster Policies](https://airflow.apache.org/docs/apache-airflow/stable/administration-and-deployment/cluster-policies.html) if you would like to do it globally

**Reply by ketozhang** [reply]
Wonderful. I didn't know this existed.
